lib/Chat/Parser/SystemMessage.php

```diff
@@ -456,6 +456,17 @@ public function parseMessage(Message $chatMessage): void {
 			if ($currentUserIsActor) {
 				$parsedMessage = $this->l->t('You deleted a reaction');
 			}
+		} elseif ($message === 'message_expire_enabled') {
+			$parsedMessage = $this->l->t('{actor} set the self-destruction timer to {seconds} seconds');
+			if ($currentUserIsActor) {
+				$parsedMessage = $this->l->t('You set the self-destruction timer to {seconds} seconds');
+			}
+			$parsedParameters['seconds'] = $parameters['seconds'];
+		} elseif ($message === 'message_expire_disabled') {
+			$parsedMessage = $this->l->t('{actor} disabled the self-destruction timer');
+			if ($currentUserIsActor) {
+				$parsedMessage = $this->l->t('You disabled the self-destruction timer');
+			}
```

Comment:
I think the same as you when I wrote this message, but I don't know what will the best to translation.

We have the problem about singular and plural and I don't know if the sequence is {timeInterval}{timeUnit} or {timeUnit}{timeInterval} in all languages.
```
{actor} set the message expiration to 1 second
{actor} set the message expiration to 1 seconds
{actor} set the message expiration to 1 day
{actor} set the message expiration to 1 days
{actor} set the message expiration to 1 week
{actor} set the message expiration to 1 weeks
```
If the order is the same in all languages, maybe we can implement this:
{actor} set the message expiration to {timeInterval} {timeUnit}

Reply:
Plurals are handled by using `$l->n()` so that's not an issue.

And then it's basically something like:
```php
<?php
$weeks = $parameters['seconds'] > (86400 * 7) ? (int) round($parameters['seconds'] / (86400 * 7)) : 0;
$days = $parameters['seconds'] > 86400 ? (int) round($parameters['seconds'] / 86400) : 0;
$hours = $parameters['seconds']  ? (int) round($parameters['seconds'] / 3600) : 0;

if ($weeks > 0) {
    $parsedMessage = $this->l->t('{actor} set the message expiration to %n week', '{actor} set the message expiration to %n weeks', $weeks);
} elseif ($days > 0) {
    $parsedMessage = $this->l->t('{actor} set the message expiration to %n day', '{actor} set the message expiration to %n days', $days);
} else {
    $parsedMessage = $this->l->t('{actor} set the message expiration to %n hour', '{actor} set the message expiration to %n hours', $hours);
}
```